Memory management and Page Counting

SpiegelSoft opened the pull request:
1. Added a PagerState.Release() statement into the constructor of the PureMemoryPager class.  
2. Added tests to ensure that the creation and disposal of pagers leaves the instance count unaffected.
3. Modified the Tree.AllPages() method to account for overflow pages.
4. Added lines into the CanAddVeryLargeValue() test to compare the results of AllPages.Count() with the PageCount state variable.
5. Corrected typo in spelling of Overflow.

SpiegelSoft committed:
1. Added a PagerState.Release() statement into the constructor of the PureMemoryPager class.  2. Added tests to ensure that the creation and disposal of pagers leaves the instance count unaffected.

SpiegelSoft committed:
Added lines that break the CanAddVeryLargeValue() test.

SpiegelSoft committed:
Modified the Tree.AllPages() method so that it accounts for overflow pages.
Renamed instances of Overlfow to Overflow.

ayende commented:
Hi,
Do you have a CLA with us?
ravendb.net/contributing

On Sun, Aug 18, 2013 at 12:08 AM, Rob Lyndon notifications@github.comwrote:

> 1. Added a PagerState.Release() statement into the constructor of the
>    PureMemoryPager class.
>    1. Added tests to ensure that the creation and disposal of pagers
>       leaves the instance count unaffected.
> 
> ---
> 
> You can merge this Pull Request by running
> 
>   git pull https://github.com/TMT-Rob/raven.voron voron
> 
> Or view, comment on, or merge it at:
> 
>   https://github.com/ayende/raven.voron/pull/2
> Commit Summary
> - 1. Added a PagerState.Release() statement into the constructor of
>   the PureMemoryPager class. 2. Added tests to ensure that the creation and
>   disposal of pagers leaves the instance count unaffected.
> 
> File Changes
> - _A_ Voron.Tests/Storage/Pagers.cshttps://github.com/ayende/raven.voron/pull/2/files#diff-0(32)
> - _M_ Voron.Tests/Voron.Tests.csprojhttps://github.com/ayende/raven.voron/pull/2/files#diff-1(1)
> - _M_ Voron.sln.DotSettings.userhttps://github.com/ayende/raven.voron/pull/2/files#diff-2(5)
> - _M_ Voron/Impl/PureMemoryPager.cshttps://github.com/ayende/raven.voron/pull/2/files#diff-3(1)
> 
> Patch Links:
> - https://github.com/ayende/raven.voron/pull/2.patch
> - https://github.com/ayende/raven.voron/pull/2.diff

SpiegelSoft commented:
I'll just print one out and send it over.

-- Rob

On 19 August 2013 11:56, Ayende Rahien notifications@github.com wrote:

> Hi,
> Do you have a CLA with us?
> ravendb.net/contributing
> 
> On Sun, Aug 18, 2013 at 12:08 AM, Rob Lyndon <notifications@github.com
> 
> > wrote:
> > 1. Added a PagerState.Release() statement into the constructor of the
> >    PureMemoryPager class.
> > 2. Added tests to ensure that the creation and disposal of pagers
> >    leaves the instance count unaffected.
> > 
> > ---
> > 
> > You can merge this Pull Request by running
> > 
> > git pull https://github.com/TMT-Rob/raven.voron voron
> > 
> > Or view, comment on, or merge it at:
> > 
> > https://github.com/ayende/raven.voron/pull/2
> > Commit Summary
> > - 1. Added a PagerState.Release() statement into the constructor of
> >   the PureMemoryPager class. 2. Added tests to ensure that the creation and
> >   disposal of pagers leaves the instance count unaffected.
> > 
> > File Changes
> > - _A_ Voron.Tests/Storage/Pagers.cs<
> >   https://github.com/ayende/raven.voron/pull/2/files#diff-0>(32)
> > - _M_ Voron.Tests/Voron.Tests.csproj<
> >   https://github.com/ayende/raven.voron/pull/2/files#diff-1>(1)
> > - _M_ Voron.sln.DotSettings.user<
> >   https://github.com/ayende/raven.voron/pull/2/files#diff-2>(5)
> > - _M_ Voron/Impl/PureMemoryPager.cs<
> >   https://github.com/ayende/raven.voron/pull/2/files#diff-3>(1)
> > 
> > Patch Links:
> > - https://github.com/ayende/raven.voron/pull/2.patch
> > - https://github.com/ayende/raven.voron/pull/2.diff
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/ayende/raven.voron/pull/2#issuecomment-22864215
> .

## 

_Rob Lyndon_
9258-8945-2998
Spotlight http://www.spotlight.com/9258-8945-2998
www.thetamingoftheshrew.co.uk
*
*
